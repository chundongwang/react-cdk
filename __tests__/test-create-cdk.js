import path from 'path';
import childProcess from 'child_process';
import dirTree from 'directory-tree';
import chalk from 'chalk';

import { generateProject, executeScript } from '../scripts/utils';

describe('Create React CDK', () => {
  
  const scriptName = path.basename(__filename, '.js');
  const cwd = path.resolve(__dirname, '../test-gen', scriptName);

  let exitCode;
  beforeAll(() => {
    /** first, we need to generate a project (into cwd folder) */
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 120000;
    return generateProject(scriptName)
      .then(code => exitCode = code)
      .then(() => new Promise((resolve, reject) => setTimeout(() => resolve(0), 100)))
  });

  it('Should finish project generation with 0', () => {
    expect(exitCode).toBe(0);
  })

  it('Should have right folder structure', () => {
    
    const projectFolder = dirTree(path.join('./test-gen', scriptName), { exclude: /node_modules/ });
    expect(projectFolder).toMatchSnapshot('test-gen/' + scriptName);
  });
});
