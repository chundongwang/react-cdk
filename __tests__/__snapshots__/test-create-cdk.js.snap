// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test-gen/test-create-cdk 1`] = `
Object {
  "children": Array [
    Object {
      "extension": "",
      "name": ".babelrc",
      "path": "test-gen/test-create-cdk/.babelrc",
      "size": 144,
      "type": "file",
    },
    Object {
      "extension": "",
      "name": ".gitignore",
      "path": "test-gen/test-create-cdk/.gitignore",
      "size": 570,
      "type": "file",
    },
    Object {
      "extension": "",
      "name": ".npmignore",
      "path": "test-gen/test-create-cdk/.npmignore",
      "size": 28,
      "type": "file",
    },
    Object {
      "children": Array [
        Object {
          "extension": ".js",
          "name": "addons.js",
          "path": "test-gen/test-create-cdk/.storybook/addons.js",
          "size": 86,
          "type": "file",
        },
        Object {
          "extension": ".js",
          "name": "config.js",
          "path": "test-gen/test-create-cdk/.storybook/config.js",
          "size": 136,
          "type": "file",
        },
      ],
      "name": ".storybook",
      "path": "test-gen/test-create-cdk/.storybook",
      "size": 222,
      "type": "directory",
    },
    Object {
      "extension": ".md",
      "name": "README.md",
      "path": "test-gen/test-create-cdk/README.md",
      "size": 80,
      "type": "file",
    },
    Object {
      "extension": ".json",
      "name": "package.json",
      "path": "test-gen/test-create-cdk/package.json",
      "size": 1066,
      "type": "file",
    },
    Object {
      "children": Array [
        Object {
          "extension": ".css",
          "name": "App.css",
          "path": "test-gen/test-create-cdk/src/App.css",
          "size": 377,
          "type": "file",
        },
        Object {
          "extension": ".js",
          "name": "Header.js",
          "path": "test-gen/test-create-cdk/src/Header.js",
          "size": 280,
          "type": "file",
        },
        Object {
          "extension": ".js",
          "name": "Header.test.js",
          "path": "test-gen/test-create-cdk/src/Header.test.js",
          "size": 235,
          "type": "file",
        },
        Object {
          "extension": ".js",
          "name": "Page.js",
          "path": "test-gen/test-create-cdk/src/Page.js",
          "size": 245,
          "type": "file",
        },
        Object {
          "extension": ".js",
          "name": "Page.test.js",
          "path": "test-gen/test-create-cdk/src/Page.test.js",
          "size": 232,
          "type": "file",
        },
        Object {
          "extension": ".svg",
          "name": "logo.svg",
          "path": "test-gen/test-create-cdk/src/logo.svg",
          "size": 2671,
          "type": "file",
        },
      ],
      "name": "src",
      "path": "test-gen/test-create-cdk/src",
      "size": 4040,
      "type": "directory",
    },
    Object {
      "extension": ".lock",
      "name": "yarn.lock",
      "path": "test-gen/test-create-cdk/yarn.lock",
      "size": 308233,
      "type": "file",
    },
  ],
  "name": "test-create-cdk",
  "path": "test-gen/test-create-cdk",
  "size": 314383,
  "type": "directory",
}
`;
